src/execution_algorithms/ltfb/truncation_selection_exchange.cpp

```diff
@@ -0,0 +1,311 @@
+  //trainer master computes trainer metric score rank/position
+  std::vector<EvalType> score_list(num_trainers);
+  if(comm.am_trainer_master()) {
+    comm.all_gather<EvalType>(score, score_list,comm.get_intertrainer_comm());
+  }
+  // Communicate trainer score list from trainer master processes
+  comm.trainer_broadcast(comm.get_trainer_master(), score_list.data(),num_trainers);
+  std::vector<EvalType> top_scores = score_list;
+  //top-k in an ascending order
+  //supports of singular metric value (for now)
+  auto met_strategy = m_metrics.begin()->second;  
+  if(low_score_wins(met_strategy)) std::sort(top_scores.begin(), top_scores.end(),std::less<EvalType>());
+  //top-k in an descending order
+  else  std::sort(top_scores.begin(), top_scores.end(),std::greater<EvalType>());
+  auto itr1 = std::adjacent_find(top_scores.begin(), top_scores.end());
+  if (itr1 != top_scores.end()) {
+    std::stringstream err;
+    err << "truncation tournament exchange currently works "
+        << "if trainers scores are unique";
+    LBANN_ERROR(err.str());
+  }
+  
+  auto itr2 = std::find(top_scores.begin(), top_scores.end(), score_list[trainer_id]);
+  auto trainer_score_pos = std::distance(top_scores.begin(),itr2);
+  
+  if(trainer_score_pos < m_truncation_k) {
+    //Winner (in top-k)
+    //for each loosing trainer
+    for(unsigned int i=m_truncation_k; i < num_trainers; i++) {
+      if(trainer_score_pos == i % m_truncation_k) {
+      //One of partners is trainer that owns score at top_scores[i]
+        auto dest = std::distance(score_list.begin(),
+                                  std::find(score_list.begin(), score_list.end(), top_scores[i]));
+  
+        auto model_string = pack(m);
+        send_string(comm, model_string, dest);
+        if(comm.am_trainer_master()) {
+          std::cout << "In LTFB TSE step " << step << ", trainer " << trainer_id << " with score " << score_list[trainer_id]; 
+          std::cout << " sends model to trainer  " << dest << " with score " << score_list[dest] << std::endl;
+        }
+      }
+
+    }
+  } else { //not in top-k, receive
+      auto src = std::distance(score_list.begin(),
+                               std::find(score_list.begin(), score_list.end(),top_scores[trainer_score_pos % m_truncation_k]));
+
+      if(comm.am_trainer_master()) { 
+        std::cout << "In LTFB TSE step " << step << ", trainer " << trainer_id << " with score " << score_list[trainer_id]; 
+        std::cout << " receives model from trainer " << src << " with score " << score_list[src] << std::endl;
+      }
+      
+      auto rcv_str = recv_string(comm, src);
+     
+
+      auto partner_model_ptr = m.copy_model();
+      auto& partner_model = *partner_model_ptr;
+      unpack(partner_model, rcv_str);
+      auto& trainer = get_trainer();
+      auto&& metadata = trainer.get_data_coordinator().get_dr_metadata();
+      m.setup(trainer.get_max_mini_batch_size(),
+             metadata,
+            /*force=*/true);
```

Comment:
```suggestion
      m.setup(trainer.get_max_mini_batch_size(),
              metadata,
              /*force=*/true);
```
Or just run `clang-format`.